lib/resty/redis.lua

```diff
@@ -82,11 +86,47 @@ function _M.connect(self, ...)
         return nil, "not initialized"
     end
 
+    local params = {...}
+
+    if not params or #params < 2 then
+        return nil, "invalid arguments"
+    end
+
     self._subscribed = false
 
-    return sock:connect(...)
-end
+    local host = params[1]
+    local connect_opts, opts, err
+    if startswith(host, "unix:") then
+        opts = params[2] or {}
+        connect_opts = {
+            pool = opts.pool,
+            poolsize = opts.poolsize,
+            backlog = opts.backlog
+        }
+        ok, err = sock:connect(host, connect_opts)
```

Comment:
I think it's better to declare `local ok, err` before using it.
For ex:
```lua
.......
local host = params[1]
local connect_opts, opts, err
local ok, err  <--
if startswith(host, "unix:") then
.......
```

Reply:
ok is already defined before at https://github.com/openresty/lua-resty-redis/blob/master/lib/resty/redis.lua#L19

Reply:
`ok` is defined before but that `ok`'s scope is larger than scope in this `_M.connect` function.
What about `err`? It seem like global variable now.
At least you should declaring the `local err`.

Reply:
`err` is also define few lines before. `luacheck` complains about shadowing vars if those are redefined again.